ignite/pkg/cosmosgen/generate_openapi.go

```diff
@@ -35,33 +37,62 @@ func generateOpenAPISpec(g *generator) error {
 		}
 	}()
 
+	specCache := cache.New[[]byte](g.cacheStorage, "generate.openapi")
+
+	anyChanges := false
+
 	// gen generates a spec for a module where it's source code resides at src.
 	// and adds needed swaggercombine configure for it.
 	gen := func(src string, m module.Module) (err error) {
-		include, err := g.resolveInclude(src)
+		dir, err := os.MkdirTemp("", "gen-openapi-module-spec")
 		if err != nil {
 			return err
 		}
+		specPath := filepath.Join(dir, "apidocs.swagger.json")
 
-		dir, err := os.MkdirTemp("", "gen-openapi-module-spec")
+		checksum, err := dirchange.ChecksumFromPaths(src, []string{m.Pkg.Path})
 		if err != nil {
 			return err
 		}
-
-		err = protoc.Generate(
-			g.ctx,
-			dir,
-			m.Pkg.Path,
-			include,
-			openAPIOut,
-		)
+		cacheKey := fmt.Sprintf("%x", checksum)
+		existing, found, err := specCache.Get(cacheKey)
 		if err != nil {
 			return err
 		}
 
+		if found {
```

Comment:
Here I am caching the content of each openapi file in the kv cache and just writing it out to disk. This helps with around 5-6 seconds on my machine on every rebuild.

Another approach I considered was to have the cache also support files and create a cache file structure so we would avoid having to recreate the files every time, but it didn't seem to matter much for performance at least.